client/components/RelatedProducts/RelatedProducts.jsx

```diff
@@ -1,23 +1,56 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import RelatedProductsList from './Related/RelatedProductsList.jsx';
 import CustomOutfitList from './Outfit/CustomOutfitList.jsx';
 import Grid from '@material-ui/core/Grid';
-import products from './../DummyData/ProductDummyData';
+import axios from 'axios';
+import config from '../../../config.js'
 
+axios.defaults.headers.common['Authorization'] = config.GITHUB_TOKEN;
 
-const RelatedProducts = (id) => {
+const RelatedProducts = ({productId}) => {
 
-  //get request for related product ids based on id of current product
-  //then get request for each product to product_id (for name, category, default_price) AND get request to id/styles (for sale_price and image) as well??
-  // id = 24156;
-  let allProducts = products.products;
-  let relatedProducts = products.relatedProducts;
-  let relatedWithNameCatPrice = allProducts.filter(product => relatedProducts.includes(product.id));
+  const [relatedProductsData, setRelatedProductsData] = useState([]);
+
+  async function getRelatedIds(productId) {
+    const relatedIdsResponse = await axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hratx/products/${productId}/related`);
+    return relatedIdsResponse.data;
+  }
+
+  useEffect(() => {
+    getRelatedIds(productId)
+    .then(relatedIdsResult => {
+
+      const productsIdGetReq = relatedIdsResult.map(id => {
+        return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hratx/products/${id}`);
+      })
+
+      const stylesGetReq = relatedIdsResult.map(id => {
+        return axios.get(`https://app-hrsei-api.herokuapp.com/api/fec2/hratx/products/${id}/styles`);
+      })
+
+      return Promise.all([...productsIdGetReq, ...stylesGetReq]);
+    })
+    .then(bothResponses => bothResponses.map(response => response.data))
+    .then(bothResponsesData => {
+
+      const midIndex = bothResponsesData.length / 2
+      const productsIdResult = bothResponsesData.slice(0, midIndex);
+      const stylesResult = bothResponsesData.slice(midIndex);
+      const combinedResults = [];
+
+      for (let i = 0; i < productsIdResult.length; i++) {
+        let temp = {...productsIdResult[i], ...stylesResult[i]};
+        combinedResults.push(temp);
+      }
+
+      setRelatedProductsData(combinedResults)
+    })
```

Comment:
I'm a bit confused what's happening here line 36-47, can you please explain @acdavitt

Reply:
@julia-thea Yeah, so basically this was the only way I could come up with to address my endless async issues yesterday.  For most of the day, I was trying to keep the array of results from the get request to products/id separate from the array of results from the request to id/styles.  But due to async issues, I ultimately had to combine the arrays in line 31.  So lines 36-47 deal with that.  Fortunately the product ids for products/ids and id/styles were at least in the same order, so by splitting the combined array in half, I could match the index and merge the objects together so that each related item now only has one object with all the info combined instead of two each.  Kind of hard to explain without seeing the data structure, but we can talk more in scrum too.  Definitely open to suggestions!

Reply:
@acdavitt thank you for the clarification! My suggestion is in the cases like this one to add code comments to a) make it easier for a person reviewing PR to understand the code, b) in case someone else has to work on this code in the future it would definitely help to improve developer experience. 
Cheers!